cpp/elasticsearch/extract_existing_fulltext_ids.sh

```diff
@@ -43,7 +55,7 @@ fi
 id_output_file=$1
 
 echo "Checking write access to output file..."
-> $id_output_file
+[ -w "$id_output_file" ] || (echo "Cannot write to to $id_output_file" && exit 1)
```

Comment:
Any reason you perform the echo and exit in a subshell?  The main shell will continue...

Reply:
Like this it is a oneliner. The main shell will also exit due to the -o errexit flag

Reply:
It's also an ugly hack and relies on the reader thinking about the errexit flag.  Generally we want code to be "obvious" and locally comprehensible, if possible.